lib/poller.js

```diff
@@ -150,37 +189,62 @@ class Poller {
   }
 
   /**
-   * Checks if secret exists, if not trigger a poll
+   * Checks status of external secret and determines time to next poll
+   * If current observed generation is older than ES generation it will poll right away
+   * otherwise check when it was last polled and set timeout for next poll
    */
-  async _checkForSecret () {
-    const kubeNamespace = this._kubeClient.api.v1.namespaces(this._namespace)
-
+  async _scheduleNextPoll () {
     try {
-      await kubeNamespace.secrets(this._secretDescriptor.name).get()
-    } catch (err) {
-      if (err.statusCode === 404) {
-        this._logger.info(`Secret ${this._secretDescriptor.name} does not exist, polling right away`)
-        this._poll()
+      const {
+        body: {
+          status: {
+            lastSync = null,
+            observedGeneration = 0
+          } = {}
+        } = {}
+      } = await this._status.get()
+
+      const currentGeneration = this._externalSecret.metadata.generation
+
+      if (observedGeneration < currentGeneration) {
+        return this._setNextPoll(0)
       }
+
+      const lastPollTime = Date.parse(lastSync) || 0
+      const elapsedTime = Date.now() - lastPollTime
```

Comment:
I didn't mean to imply we should remove it.  I like the defensive behavior of scheduling it immediately (and then hopefully getting a more sane lastPollTime value).  Maybe bring the code back and add a comment?

Reply:
Ah yes, must have been tired last night. I was thinking this became a non issue after the changes but a negative elapsedTime here gives a funny next poll time 😅 I'll add it back.